test/controllers/GuidanceControllerSpec.scala

```diff
@@ -25,162 +25,206 @@ import play.api.test.FakeRequest
+    val questionView = app.injector.instanceOf[views.html.question_page]
+    val target = new GuidanceController(errorHandler, 
+                                             view, 
+                                             questionView, 
+                                             new NextPageFormProvider(), 
+                                             mockGuidanceService, 
+                                             stubMessagesControllerComponents())
+  }
+
+
+  "Calling a valid URL path to a Question page in a process" should {
+
+    "return an OK response" in new QuestionTest {
+      val result: Future[Result] = target.getPage(relativePath)(fakeRequest)
+      status(result) mustBe Status.OK
+    }
+
+    "be a HTML response" in new QuestionTest {
+      val result: Future[Result] = target.getPage(relativePath)(fakeRequest)
+      contentType(result) mustBe Some("text/html")
+    }    
+
+  }
+
+  "Submitting a blank Question page form" should {
+
+    "return a BadRequest response" in new QuestionTest {
+      val result: Future[Result] = target.submitPage(relativePath)(fakeRequest)
+      status(result) mustBe Status.BAD_REQUEST
+    }
+  }
+
+  "Submitting an answered Question page form" should {
+
+    "return a SeeOther response" in new QuestionTest {
+      override val fakeRequest = FakeRequest("GET", path).withSession(SessionKeys.sessionId -> processId).withFormUrlEncodedBody(("url" -> "/hello")).withCSRFToken
+      val result: Future[Result] = target.submitPage(relativePath)(fakeRequest)
+      status(result) mustBe Status.SEE_OTHER
+    }
+  }
+
+  trait ScratchTest extends MockGuidanceService {
+    lazy val uuid = "683d9aa0-2a0e-4e28-9ac8-65ce453d2730"
+    lazy val fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("GET", "/")
+    val expectedUrl = "/start-url"
+
+    private lazy val errorHandler = app.injector.instanceOf[config.ErrorHandler]
+    private lazy val view = app.injector.instanceOf[views.html.standard_page]
+    private lazy val questionView = app.injector.instanceOf[views.html.question_page]
+
+    lazy val target = new GuidanceController(errorHandler, 
+                                             view, 
+                                             questionView, 
+                                             new NextPageFormProvider(), 
+                                             mockGuidanceService, 
+                                             stubMessagesControllerComponents())
+    
+  }
 
-    trait ScratchTest extends MockGuidanceService {
-      lazy val uuid = "683d9aa0-2a0e-4e28-9ac8-65ce453d2730"
-      lazy val fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("GET", "/")
-      val expectedUrl = "/start-url"
+  "Calling the scratch process endpoint with a valid UUID" should {
 
+    trait ScratchTestWithValidUUID extends ScratchTest {
       MockGuidanceService
         .scratchProcess(uuid, uuid)
```

Comment:
I would consider creating a dummy repository id here. Although it is not necessary reusing the uuid might confuse some people.